ostd/libs/linux-bzimage/builder/src/encoder.rs

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module is used to compress kernel ELF.
+
+use std::io::Write;
+
+use libflate::{deflate, gzip, zlib};
+
+pub fn compress_kernel(kernel: &[u8]) -> Vec<u8> {
+    match std::env::var("COMPRESSION_FORMAT") {
```

Comment:
I think that the compression format can be exposed as a OSDK argument rather than specifying as an environment varariable. Checkout https://github.com/asterinas/asterinas/blob/main/osdk/src/config/scheme/action.rs about how to add a configure option to OSDK.toml. And checkout https://github.com/asterinas/asterinas/blob/main/osdk/src/cli.rs about how to add a CLI option to cargo-osdk.